license-tool/src/main.rs

```diff
@@ -0,0 +1,378 @@
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "spdx-id")]
+struct SpdxIdArgs {
+    /// the package name used to look up for the licenses
+    #[argh(option, short = 'p')]
+    package_name: String,
+}
+
+/// Creates a copy of the licenses files in the dest directory
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "fetch")]
+struct FetchArgs {
+    /// the destination folder for the licenses
+    #[argh(option, short = 'd')]
+    destination: PathBuf,
+}
+
+/// Prints out a space-separated list of the paths to the licenses files
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "path")]
+struct PathArgs {
+    /// the package name used to look up for the licenses
+    #[argh(option, short = 'p')]
+    package_name: String,
+    /// the source folder where the licenses are
+    #[argh(option, short = 's')]
+    source: PathBuf,
+}
+
+/// Holds the configurations for package's licenses
+#[derive(Deserialize, Debug)]
+struct PackageLicense {
+    #[serde(rename(deserialize = "spdx-id"))]
+    spdx_id: String,
+    #[serde()]
+    licenses: Vec<License>,
+}
+
+/// Holds the configurations for a license
+#[derive(Deserialize, Debug, Clone)]
+struct License {
+    #[serde(rename(deserialize = "license-url"))]
+    license_url: Url,
+    #[serde()]
+    path: Option<String>,
+}
+
+/// Prints the spdx id for the package
+fn print_spdx_id<S>(packages_licenses: &HashMap<String, PackageLicense>, package: S) -> Result<()>
+where
+    S: AsRef<str>,
+{
+    let package = package.as_ref();
+    let package_license = packages_licenses.get(package).context(format!(
+        "Couldn't find configuration for package '{}'",
+        package
+    ))?;
+    println!("{}", package_license.spdx_id);
+    Ok(())
+}
+
+/// Prints a space separated list of paths
+fn print_paths<S, P>(
+    packages_licenses: &HashMap<String, PackageLicense>,
+    package_name: S,
+    source: P,
+) -> Result<()>
+where
+    S: AsRef<str>,
+    P: AsRef<Path>,
+{
+    let package_name = package_name.as_ref();
+    let package_license = packages_licenses.get(package_name).context(format!(
+        "Couldn't find configuration for package '{}'",
+        package_name
+    ))?;
+    println!("{}", get_paths(package_license, source)?.join(" "));
+    Ok(())
+}
+
+/// Fetches all the licenses for the passed map of package licenses
+async fn fetch_all_licenses<P>(
+    packages_licenses: &HashMap<String, PackageLicense>,
+    dest: P,
+) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    for package_license in packages_licenses.values() {
+        fetch_licenses(package_license, &dest).await?;
+    }
+
+    Ok(())
+}
+
+/// Fetches the licenses in the `PackageLicense` object, and creates a copy of them in `dest`
+async fn fetch_licenses<P>(package_license: &PackageLicense, dest: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    for license in &package_license.licenses {
+        let path: PathBuf = get_path(&license, &dest)?.into();
```

Comment:
```suggestion
        let path: PathBuf = get_path(license, &dest)?.into();
```
clippy says this is a needless borrow (same with lines 188, 298, and 309).